File: GestiPro.API/Models/Transacao.cs
namespace GestiPro.API.Models;

public class Transacao
{
    public int Id { get; set; }
    public int IdCategoria { get; set; }
    public int IdPessoa { get; set; }
    public string Descricao { get; set; }
    public decimal Valor { get; set; }
    public TipoTransacao Tipo { get; set; }
    public Categoria Categoria { get; set; }
    public Pessoa Pessoa { get; set; }
}

public enum TipoTransacao
{
    Despesa = 0,
    Receita = 1
}

File: GestiPro.Web/src/types/transacao.ts
// Enum dos tipos de transacao pro backend
export enum TipoTransacao {
  Despesa = 0,
  Receita = 1,
}

export const TipoTransacaoLabel: Record<TipoTransacao, string> = {
  [TipoTransacao.Despesa]: 'Despesa',
  [TipoTransacao.Receita]: 'Receita',
};

// Retorno transacao
export interface Transacao {
  id: number;
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  tipoDescricao: string;
  idCategoria: number;
  categoriaDescricao: string;
  idPessoa: number;
  pessoaNome: string;
}

// Dados enviados para a api da transacao
export interface TransacaoRequest {
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  idCategoria: number;
  idPessoa: number;
}
